components/PostPreviewsGrid/index.tsx

```diff
@@ -0,0 +1,62 @@
+import CustomButton from "components/CustomButton";
+import { IPost } from "components/Post";
+import PostPreview from "components/PostPreview";
+import { useEffect, useState } from "react";
+
+interface IGridPostPreview {
+  posts: Array<IPost>;
+  loadMoreButton?: boolean;
+}
+
+const PostPreviewsGrid = ({ posts, loadMoreButton }: IGridPostPreview) => {
+  const [currentPostsDisplayed, setCurrentPostsDisplayed] = useState<number>(6);
+  const [currentPosts, setCurrentPosts] = useState<IPost[]>(posts.slice(0, 6));
+
+  useEffect(() => {
+    setCurrentPosts(posts.slice(0, currentPostsDisplayed));
+  }, [currentPostsDisplayed, posts]);
+
+  const onLoadMore = () => {
+    setCurrentPostsDisplayed(currentPostsDisplayed + 6);
+  };
+
+  return (
+    <>
+      <div className="postPreviewGrid">
+        <div className="PostPreviewsGridCards">
+          {currentPosts.map((post, key) => (
+            <PostPreview
+              variant="card"
+              post={post}
+              key={`post-preview-card-${key}`}
+            />
+          ))}
+        </div>
+        {loadMoreButton ? (
+          <CustomButton label="Load More" onClick={onLoadMore} />
+        ) : null}
```

Comment:
One more case which would be interesting to address:

When we filter, we might not want to load more posts. Can you disable this button if a filter is selected?

Reply:
Yes. On the other hand, when there are not more post to display, the button should disappear . I will add this case as well.